hail/python/hailtop/aiogoogle/client/compute_client.py

```diff
@@ -1,29 +1,33 @@
+import copy
 import uuid
-from typing import Mapping, Any, Optional, MutableMapping
+import asyncio
+from typing import Mapping, Any, Optional, MutableMapping, List, Dict
 import logging
 
 from .base_client import BaseClient
-from hailtop.utils import sleep_and_backoff
+from hailtop.utils import retry_transient_errors
 
 log = logging.getLogger('compute_client')
 
 
-async def request_with_wait_for_done(request_f, path, params: MutableMapping[str, Any] = None, **kwargs):
-    assert 'params' not in kwargs
+class GCPError:
+    def __init__(self, code, message):
+        self.code = code
+        self.message = message
```

Comment:
Can we maybe call this GCPErrorInfo? I'm a bit worried the name looks like it should be raise-able.

Reply:
Let's put types on these fields. Without the types, `code` sounds like it should be an int.